src/transformers/trainer.py

```diff
@@ -525,6 +528,11 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
         if not isinstance(self.train_dataset, collections.abc.Sized):
             return None
 
+        generator = None
+        if self.args.world_size <= 1 and _is_torch_generator_available:
+            generator = torch.Generator()
+            generator.manual_seed(int(torch.empty((), dtype=torch.int64).random_().item()))
```

Comment:
i'm very late here, but shouldn't this use `self.args.seed`?

Reply:
No, this is copied from PyTorch actually. Since torch has been seeded, this will be deterministic.